terms/onlinelearning_df.tex

```diff
@@ -0,0 +1,45 @@
+\chapter{Online Learning Algorithms}
+\label{ch:online-learning}\index{online-learning}
+
+\section{Introduction}
+
+\begin{figure}
+  \includegraphics{online_learning/traditional.png}
+  \caption{The traditional batch train-test machine learning approach workflow.}
+  \label{fig:traditional_ml}
+\end{figure}
+
+In the traditional machine learning approach depicted in figure~\ref{fig:traditional_ml}, we usually have some historical data to train an algorithm on for predicting some future events \citep{oza_online_2005}. However, since most data environments are dynamic and will change, the trained model eventually becomes outdated. To tackle this, we usually automate model re-training based on a timeframe (i.e. weekly or daily basis). Although this helps with keeping the model up-to-date, this is still not enough. Even if we consider model re-training on a daily basis, the model would still be at least one day late. To further explain this, let us consider the case study of a recommender engine for an online portal where it shows users items based on their respective preferences. If we consider the situation that one specific item has gone globally viral overnight, our recommender system will fail to display this item to the whole population and instead would show the item to a part of the user-base.
```

Comment:
Figure should start in capital in this case as you are referring to a specific figure.  Also, see point 28 in the guidelines.